test/suites-ostree/upgrade.robot

```diff
@@ -0,0 +1,88 @@
+*** Settings ***
+Documentation       Tests related to MicroShift upgradeability on OSTree-based systems
+
+Resource            ../resources/common.resource
+Resource            ../resources/ostree.resource
+Library             Collections
+
+Suite Setup         Setup
+Suite Teardown      Teardown
+
+Test Tags           ostree
+
+
+*** Variables ***
+${USHIFT_HOST}      ${EMPTY}
+${USHIFT_USER}      ${EMPTY}
+
+
+*** Test Cases ***
+Upgrade
+    [Documentation]    Perform clean upgrades
+    [Template]    Perform And Verify Upgrade
+    microshift/413    microshift/local    # local = RPM from source
+    # microshift/main    microshift/local
+    # microshift/current-nightly    microshift/main
+
+
+*** Keywords ***
+Setup
+    [Documentation]    Test suite setup
+    Check Required Env Variables
+    Login MicroShift Host
+
+Teardown
+    [Documentation]    Test suite teardown
+    Logout MicroShift Host
+
+Perform And Verify Upgrade
+    [Documentation]    Template for performing a clean upgrade from one ref to another
+    ...    and verifying it was successful
+    [Arguments]    ${initial_ref}    ${target_ref}    ${cleanup}=${TRUE}
```

Comment:
This keyword doesn't care what initial state of the machine is - it'll clean up microshift, deploy 1st commit, then another.
It might not be what we want to some scenarios, but it might be useful for periodic and is definitely useful for development (no need to manually go back to initial commit)

Reply:
Where does it get those values? Is it possible to make the test suite take inputs for the origin and destination image references so we can reuse it for multiple scenarios?

Reply:
This keyword is used as a "test template" by test case `Upgrade`. Each line in the test case `Upgrade` contains args passed to this keyword.
https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#test-templates

I'm not sure if we'll use it like this. If we can spawn N VMs with desired initial ref, then we can cut the part about deploying initial ref but it's very very useful for local development